controllers/storagecluster/cephblockpools_test.go

```diff
@@ -28,11 +28,13 @@ func TestCephBlockPools(t *testing.T) {
 		cp := &Platform{platform: eachPlatform}
 		for _, c := range cases {
 			var objects []runtime.Object
-			if c.createRuntimeObjects {
-				objects = createUpdateRuntimeObjects(cp)
-			}
 			t, reconciler, cr, request := initStorageClusterResourceCreateUpdateTestWithPlatform(
 				t, cp, objects)
+			if c.createRuntimeObjects {
+				objects = createUpdateRuntimeObjects(cp, reconciler)
+				t, reconciler, cr, request = initStorageClusterResourceCreateUpdateTestWithPlatform(
+					t, cp, objects)
+			}
```

Comment:
I'm not quite sure I understand the need for this change... how does it relate to the platform detection changes? Also, why are you running `initStorageClusterResourceCreateUpdateTestWithPlatform()` twice? That seems like a waste. Either you run it with or without the new objects. It seems you should just change the one line to add the new parameter.